controllers/pravegacluster_controller.go

```diff
@@ -388,11 +388,29 @@ func (r *PravegaClusterReconciler) reconcileControllerService(p *pravegav1beta1.
 
 	service := MakeControllerService(p)
 	controllerutil.SetControllerReference(p, service, r.Scheme)
-	err = r.Client.Create(context.TODO(), service)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return err
-	}
+	currentService := &corev1.Service{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: service.Name, Namespace: p.Namespace}, currentService)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Client.Create(context.TODO(), service)
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
 
+		} else {
+			return err
+		}
+
```

Comment:
(nitpicking) Could you remove the extra empty lines from the new code to conform to the file formatting? Thanks!

Reply:
Done